libsolidity/codegen/ContractCompiler.cpp

```diff
@@ -983,3 +996,26 @@ eth::AssemblyPointer ContractCompiler::cloneRuntime() const
 	a << u256(0x20) << u256(0) << Instruction::RETURN;
 	return make_shared<eth::Assembly>(a);
 }
+
+void ContractCompiler::popScopedVariables(ASTNode const* _node)
+{
+	unsigned blockHeight = m_scopeStackHeight[m_modifierDepth][_node];
+	unsigned stackDiff = m_context.stackHeight() - blockHeight;
+	CompilerUtils(m_context).popStackSlots(stackDiff);
+	m_context.removeVariablesAboveStackHeight(blockHeight);
+	m_scopeStackHeight[m_modifierDepth].erase(_node);
+	if (m_scopeStackHeight[m_modifierDepth].size() == 0)
+		m_scopeStackHeight.erase(m_modifierDepth);
+}
+
+void ContractCompiler::visitLoop(BreakableStatement const* _loop)
+{
+	m_scopeStackHeight[m_modifierDepth][_loop] = m_context.stackHeight();
+}
+
+void ContractCompiler::popAndJump(unsigned _amount, eth::AssemblyItem const& _jumpTo)
```

Comment:
This might be better suited inside the context itself.

Reply:
What about inside `CompilerUtils` instead of the context?